synapse/storage/room.py

```diff
@@ -541,7 +543,14 @@ def _store_content_index_txn(self, txn, event, key):
     def _update_retention_policy_for_room_txn(self, txn, event):
         if (
             hasattr(event, "content")
-            and ("min_lifetime" in event.content or "max_lifetime" in event.content)
+            and (
```

Comment:
this leaves open the possibility that `min_lifetime` is valid and `max_lifetime` is bogus.